File: week0/codefiles/cash.py
# realpython.com/python-strings/
# docs.python.org/3.1/library/string.html

#quarters (25¢), dimes (10¢), nickels (5¢), and pennies (1¢)
cents = 0
def main():
    valUser = get_input("Change owed: ")
    cents = round(valUser * 100)
    #Calculate quarters and remainer change
    quarter = cents//25
    cents = cents % 25
    #Calculate dimes and remaining change
    dimes = cents//10
    cents = cents % 10
    #Calculate Nickels and remaining change
    nickels = cents//5
    cents = cents % 5
    #Number of pennies is the remaining change
    totalCoins = quarter + dimes + nickels + cents
    print(str(totalCoins))

# stackoverflow.com/questions/35808484/how-to-check-if-a-string-represents-a-float-number
def get_input(prompt):
    while True:
        userIp = input(prompt)
        try:
            if not userIp == '' and  float(userIp) and float(userIp) > 0:
                break
        except ValueError:
          continue

    return float(userIp+'\n')

if __name__ == "__main__":
    main()

File: week0/codefiles/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void){

    float fltChange;
    bool isValid = false;

	do
	{
		fltChange = get_float("Change: ");
		if (fltChange > 0 )
        {
			isValid = true;
		}
	}
	while (!isValid);

    if (isValid)
    {
        int intCoins = 0;
        int intDimes = 0;
        int intNickel = 0;
        int intPennie = 0;

		int iRemainerChange = 0;
        int intCents = round(fltChange * 100);

		//Calculate Number of Quarters
        int intQuarter = (intCents / 25);

		//Calculate Number of Dimes
		iRemainerChange = intCents - (intQuarter * 25);
        if(intCents % 25 != 0){
			intDimes = (iRemainerChange / 10);
		}

		//Calculate Number of Nickel
		iRemainerChange = iRemainerChange - (intDimes * 10);
		if(iRemainerChange % 10 != 0)
		{
			intNickel =  (iRemainerChange / 5);
		}

		//Calculate Number of Pennies
		intPennie =  iRemainerChange - (intNickel * 5);

		//Calculate and show result
		intCoins = intQuarter + intDimes + intNickel + intPennie;
		printf("%i\n", intCoins);
	}
}